### PR TITLE
DOCS: Improve Internationalization guide

### LINKED ISSUE #12363: Internationalization doc mentions deprecated transifex-client
### Describe the bug

In the [Internationalization](https://www.sphinx-doc.org/en/master/usage/advanced/intl.html) page, the [Using Transifex service for team translation](https://www.sphinx-doc.org/en/master/usage/advanced/intl.html#using-transifex-service-for-team-translation) section tells to use the Python-based transifex-client, however it is deprecated. Quoting the info on the new  tool:

> Please start using the new Transifex CLI [here](https://github.com/transifex/cli) , since this software is considered deprecated (as of January 2022) and will sunset on Nov 30, 2022.

The new Transifex tool is written in Go, commonly named "Transifex CLI tool" , that has almost the same options (tx pull, tx push, etc.) and has the same binary name `tx`. The installation and usage instructions can be found in the documentation (https://developers.transifex.com/docs/cli) and source code repository (https://github.com/transifex/cli)

The section needs to be updated with this information.

### How to Reproduce

It doesn't apply to the issue

### Environment Information

```text
It doesn't apply to the issue
```

### Sphinx extensions

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rffontenelle**
### Feature or Bugfix

- Refactoring

### Purpose
Fix and update information ins this page.

### Detail
- Documentation needs to mention the new Transifex CLI tool instead of the deprecated transifex-client
- 

### Relates
- Fixes #12363

### CODE REVIEW

**Comment by chrisjsewell** [comment]
Hi @rffontenelle can you confirm this ready for review/merge, as you've made a few changes since opening 😅  (best to set as a Draft until it is ready)

**Comment by rffontenelle** [comment]
@chrisjsewell Hi, I believe I'm satisfied for the moment. If I come up with other improvements, I'll open another PR in the proper time. Could you please review?

File: doc/usage/advanced/intl.rst [hunk 1]
```diff
@@ -218,69 +218,103 @@ Using Transifex service for team translation
...
    your project name.
 
    For example:
 
+   :Organization ID: ``sphinx-document``
    :Project ID: ``sphinx-document-test_1_0``
    :Project URL: ``https://www.transifex.com/projects/p/sphinx-document-test_1_0/``
 
-#. Create config files for :command:`tx` command.
+#. Create a API token to be used in the command-line.
```
Comment (chrisjsewell) [inline doc/usage/advanced/intl.rst]:
```suggestion
#. Create an API token to be used in the command-line.
```

File: doc/usage/advanced/intl.rst [hunk 2]
```diff
@@ -218,69 +218,103 @@ Using Transifex service for team translation
...
    .. code-block:: console
 
-      $ pip install transifex-client
+      $ curl -o- https://raw.githubusercontent.com/transifex/cli/master/install.sh | bash
 
    .. seealso:: `Transifex Client documentation`_
 
-#. Create your Transifex_ account and create new project for your document.
+#. Create your Transifex_ account and create new project and a organization
+   for your document.
```
Comment (chrisjsewell) [inline doc/usage/advanced/intl.rst]:
```suggestion
#. Create your Transifex_ account and create a new project and an organization
   for your document.
```

**Review by chrisjsewell — APPROVED** [review]
great cheers!

**Pull request merged**